core/kernel/wait_queue.c

```diff
@@ -69,24 +74,45 @@ void wq_wait_init_condvar(struct wait_queue *wq, struct wait_queue_elem *wqe,
 	cpu_spin_unlock_xrestore(&wq_spin_lock, old_itr_status);
 }
 
-void wq_wait_final(struct wait_queue *wq, struct wait_queue_elem *wqe,
-		   const void *sync_obj, const char *fname, int lineno)
+static uint32_t wq_wait_final_helper(struct wait_queue *wq,
+				     struct wait_queue_elem *wqe,
+				     uint32_t tmo, const void *sync_obj,
+				     const char *fname, int lineno)
 {
 	uint32_t old_itr_status;
 	unsigned done;
+	uint32_t ret;
 
 	do {
-		do_notif(notif_wait, wqe->handle,
+		ret = do_notif(notif_wait_timeout, wqe->handle, tmo,
 			 "sleep", sync_obj, fname, lineno);
 
 		old_itr_status = cpu_spin_lock_xsave(&wq_spin_lock);
 
+		if (ret == TEE_ERROR_BUSY) /* timed-out */
```

Comment:
Wouldn't `TEE_ERROR_TIMEOUT` be a better match?

Reply:
Ok, I didn't notice this code, it would be more suitable for this scenario.
I will change to use `TEE_ERROR_TIMEOUT` in op-tee and linux kernel